applications/DEMApplication/python_scripts/DEM_analysis_stage.py

```diff
@@ -330,71 +331,98 @@ def GetProblemNameWithPath(self):
         return os.path.join(self.mdpas_folder_path, self.DEM_parameters["problem_name"].GetString())
 
     def GetMpFilename(self):
-        return self.GetProblemNameWithPath() + "DEM"
+        return 'DEM'
```

Comment:
It seems like this method is no longer doing what it says.

Reply:
Well, in a way yes. The only part identifying the name is 'DEM', the rest is common to all the other mp. We could update the function name to 'GetMpFileTag', perhaps

Reply:
I am more for conserving the old behavior if possible, so in some cases we can derive the stage and make it return a file name completely different from the others. I mean different tag and different 'main name'.

Reply:
Please, take a look at the new functions provided.